lib/puppet/type/tidy.rb

```diff
@@ -222,7 +222,16 @@ def initialize(hash)
   # Make a file resource to remove a given file.
   def mkfile(path)
     # Force deletion, so directories actually get deleted.
-    Puppet::Type.type(:file).new :path => path, :backup => self[:backup], :ensure => :absent, :force => true
+    parameters = {
+      :path => path, :backup => self[:backup],
+      :ensure => :absent, :force => true
+    }
+
+    if !self[:noop].nil?
```

Comment:
I'd prefer `unless` here rather than `if !`

Reply:
ah yeah, let me simplify that to just `if self[:noop]` to avoid double negatives.